tools/nuget/README.md

```diff
@@ -0,0 +1,31 @@
+# CSFML NuGet Packages
+This directory contains all the files needed to build and package the native CSFML libraries for [SFML.Net](github.com/SFML/SFML.Net).
+**All scripts must be run from the directory they are located in**. 
+That means that your current working directory (`$PWD`) must be the same where the script is.
+
+All build scripts follow the same routine:
+1. Inside of this folder, create a subdirectory `Build/(RID)` to use as a working directory, where `(RID)` is the [Runtime Identifier](https://docs.microsoft.com/en-us/dotnet/core/rid-catalog) for the target of the libraries being built
+    * If the folder already exists, it is emptied
+2. Configure & compile SFML with cmake inside of a `SFML` subdirectory
+3. Configure & compile CSFML with cmake inside of a `CSFML` subdirectory
+4. Copy the resulting libraries to the NuGet package
+    * For example, `libcsfml-window.so` will be copied to `/CSFML/runtimes/(RID)/native/libcsfml-window.so`
+
+
+* `build.win.ps1`
+    * A [PowerShell](https://github.com/PowerShell/PowerShell) script to build Windows native libraries. By default, both `win-x86` and `win-x64` are built. To build only one specific Runtime Identifier, pass it to the script as a parameter named `RID` (`./build.win.ps1 -RID win-x64`)  
+    You must have Visual Studio 2017, cmake, and the Windows SDK 8.1 installed to run this script.
+
+* `build.linux.sh`
+    * A [Bash](https://www.gnu.org/software/bash/) script to build libraries for a generic Linux distribution. The [Runtime Identifier](https://docs.microsoft.com/en-us/dotnet/core/rid-catalog) of the linux distribution must be specified as a parameter to the script.  
+    This generic script doesn't install any of its dependencies. As such, you must have cmake and make installed, as well as all [SFML Dependencies](https://www.sfml-dev.org/tutorials/2.5/compile-with-cmake.php#installing-dependencies) available to cmake.
+
+* `build.alpine-x64.sh`, `build.debian-x64.sh`, `build.fedora-x64.sh`
```

Comment:
Would it be possible to extend these scripts to automate the build-up and tear-down of the docker containers?

Reply:
What permission issues are there when binding the volume?

Binding the whole repo seems to make the most sense to me, as we don't really want to clone the repository again, plus at one point you need to bind something to the host anyway or use `scp` or similar to get the assemblies out of the docker image.

Right now, my main issue, that I don't exactly know how to start the various docker containers and what additional commands need to be executed before I can run the build scripts.

Reply:
On Linux, every file or directory created by the container will be owned by root, which can cause permission errors when deleting or moving them.

On Windows, host files/directories are mounted with fixed permissions and ownership, which can't be changed. This may cause issues if certain programs require specific permissions (the same way that ssh requires specific permissions for your keys). Also, the host permissions of files created inside the container are completely dependent on implementation details of Docker itself.

Bind-mounting the repo is also the solution I would go with, but it's good to keep possible permission issues in mind.

Reply:
I've never had permission issues with binding volumes, I've even bound locally mounted samba shares and it's working just fine. Then again my Docker host was always a Linux distro.

Wouldn't the idea here be to startup a new docker container, install what's needed, build SFML, build CSFML, copy assemblies and then shutdown/remove the container again? That way I don't see where a non-root user would want to copy/access anything or am I missing something here?

As for the Linux Docker > Windows host copy, not sure what kind of permission is added, but I'd say that regardless of the situation, the nuget building isn't something everyone on every possible system should be running, as such it's okay to expect that the user running it, has all the required rights etc.

Reply:
Yeah, I understand, we shouldn't have any issues in the case of the NuGet package. I'll expand the scripts to create the Docker containers by bind-mounting the repo.

Reply:
Thanks 🙂